### ISSUE TITLE
Cannot download model "'inswapper_128.onnx'"

### ISSUE THREAD

**Issue by Southmelon-Allen** [deepinsight/insightface#2294]
swapper = insightface.model_zoo.get_model('inswapper_128.onnx', download=True, download_zip=True)

cannot download the model from given url. 
RuntimeError: Failed downloading url https://github.com/deepinsight/insightface/releases/download/v0.7/inswapper_128.onnx.zip

**Comment by Southmelon-Allen** [comment]
cannot download zip file manually through this link, Github respond "Not Found"

**Comment by zhouzhengguang** [comment]
+1

**Comment by nttstar** [comment]
The swapping demo had been moved to discord bot.

**Comment by zhouzhengguang** [comment]
@nttstar 
That's awesome! But, I want the inswapper_128.onnx model file to test my photo locally, can you share the model on drive or somewhere? Many thanks!

**Comment by nttstar** [comment]
The 128 model does not perform as well as the Discord version. At present, there is no reason to upload it again.

**Comment by liyaodong** [comment]
But seems the discord app is down at the moment. In the meantime inswapper_128 is deleted. Seem like no easy way to try the face swap feature by now.

BTW, is the discord version swap model will be public in the future?

Thanks anyway.

**Comment by nttstar** [comment]
> But seems the discord app is down at the moment

No, it's still in service.

> BTW, is the discord version swap model will be public in the future?

Not very likely

**Comment by YuanBo66668888** [comment]
还有其他办法么？ @nttstar

**Comment by nihs269** [comment]
Thanks @nttstar!! Nothing is secret for me =))

**Comment by Tylersuard** [comment]
Ok but the discord service only allows 50 commands per day.  When the face swapping only works well 1 in 50 times, that's not a lot.  I want to run this locally so I can face swap more and get some quality results.  How can I do that?

**Comment by mike-hearn** [comment]
Out of curiosity, is there a commercial goal by privatizing this model? I haven't used the Discord. Do you have to pay for it?

It feels like there must be some ulterior motive surrounding removing the 128x128 model, and refusing to release any larger models.

I'm not opposed to you making money off your work obviously, but I just want to be clear what the rationale is.

**Comment by dai-shuo** [comment]
> The 128 model does not perform as well as the Discord version. At present, there is no reason to upload it again.

it's reasonable for you guys to close source. 
but it'd be nicer to remove the inswapper from the opensourced python package since it's not working anymore.

**Comment by CGMikeG** [comment]
awwww thats painful would be nice if we can get if we can use this locally

**Comment by nihs269** [comment]
Please inbox me to receive the similar model like "inswapper_128"

**Comment by saurabhaloneai** [comment]
Link : https://drive.google.com/file/d/1krOLgjW2tAPaqV-Bw4YALz0xT5zlb5HF/view { thanks to that reddit guy }

**Comment by Fibonacci134** [comment]
> Out of curiosity, is there a commercial goal by privatizing this model? I haven't used the Discord. Do you have to pay for it?
> 
> It feels like there must be some ulterior motive surrounding removing the 128x128 model, and refusing to release any larger models.
> 
> I'm not opposed to you making money off your work obviously, but I just want to be clear what the rationale is.

They basically got greedy, released the model and then pulled it. Expected better from them. Too bad the code and model are out already so they cant really do anything about it now.

**Comment by SolessChong** [comment]
> uld be nice if we can

People have their rights to pull the model. I appreciate what the left opensourced. 



> Link : https://drive.google.com/file/d/1krOLgjW2tAPaqV-Bw4YALz0xT5zlb5HF/view { thanks to that reddit guy }

And I appreciate this even more.

**Comment by Dsgathebest** [comment]
> Enlace: https://drive.google.com/file/d/1krOLgjW2tAPaqV-Bw4YALz0xT5zlb5HF/view {gracias a ese tipo de reddit}

guy, do you now where to save this file? i have a problem with that

**Comment by wiesbaden6200** [comment]
How can I run this locally?
or how can I forward it to a different address in colab.

"inswapper_128.onnx" 

Thanks

**Comment by grahamquest** [comment]
> > uld be nice if we can
> 
> People have their rights to pull the model. I appreciate what they left opensourced.
> 
> > Link : https://drive.google.com/file/d/1krOLgjW2tAPaqV-Bw4YALz0xT5zlb5HF/view { thanks to that reddit guy }
> 
> And I appreciate this even more.

Agreed, and VERY MUCH agreed for that 2nd part. Some people are different, some see chaos and misuse in their projects and its fine for them to do whatever they want with it. As for me? I learned to BACK UP files a lot more.

**Comment by shandowsoftware** [comment]
> 并且非常同意第二部分。有些人是不同的，有些人在他们的项目中看到混乱和滥用，他们可以用它做任何他们想做的事情。至于我呢？我学会了更多备份文件。



> > > uld be nice if we can
> > 
> > 
> > People have their rights to pull the model. I appreciate what they left opensourced.
> > > Link : https://drive.google.com/file/d/1krOLgjW2tAPaqV-Bw4YALz0xT5zlb5HF/view { thanks to that reddit guy }
> > 
> > 
> > And I appreciate this even more.
> 
> Agreed, and VERY MUCH agreed for that 2nd part. Some people are different, some see chaos and misuse in their projects and its fine for them to do whatever they want with it. As for me? I learned to BACK UP files a lot more.



> Link : https://drive.google.com/file/d/1krOLgjW2tAPaqV-Bw4YALz0xT5zlb5HF/view { thanks to that reddit guy }

this is the real  open source,some people are not open enough

**Comment by stasha17** [comment]
> Link : https://drive.google.com/file/d/1krOLgjW2tAPaqV-Bw4YALz0xT5zlb5HF/view { thanks to that reddit guy }

Is there any reason why there are different inswapper_128.onnx files? This one at the google drive is 541mb and the other one I have is 182mb.  Is there a difference in the files, and if so, what?

**Comment by Maxencehabar** [comment]
> How can I run this locally? or how can I forward it to a different address in colab.
> 
> "inswapper_128.onnx"
> 
> Thanks

Did you get the answer ? I'm trying to do the same thing and I'm having hard times trying.
Thanks!

**Comment by Mayorc1978** [comment]
> > Link : https://drive.google.com/file/d/1krOLgjW2tAPaqV-Bw4YALz0xT5zlb5HF/view { thanks to that reddit guy }
> 
> Is there any reason why there are different inswapper_128.onnx files? This one at the google drive is 541mb and the other one I have is 182mb. Is there a difference in the files, and if so, what?

I would like to know as well, I have the 5XX MB version too, where did you get the 182mb version?
Can you share it?
By the way, what is the official version size? If the 182mb version is like a pruned version and works precisely the same it could be better for memory consumption.

**Comment by Ubermeisters** [comment]
So there's also this, just FYI everyone.

https://github.com/facefusion/facefusion-assets/releases/download/models/inswapper_128.onnx

**Comment by qym77** [comment]
> Link : https://drive.google.com/file/d/1krOLgjW2tAPaqV-Bw4YALz0xT5zlb5HF/view { thanks to that reddit guy }

after download how to use it? i am watching Rob Mulla'video, but it can't work as the video,i think i put it into the right folder,but it from the code seems not find it? plz!

**Comment by qym77** [comment]
> > How can I run this locally? or how can I forward it to a different address in colab.
> > "inswapper_128.onnx"
> > Thanks
> 
> Did you get the answer ? I'm trying to do the same thing and I'm having hard times trying. Thanks!

AssertionError                            Traceback (most recent call last)
Cell In[36], line 1
----> 1 swapper = insightface.model_zoo.get_model('inswapper_128.onnx', 
      2                                           download=False, 
      3                                           download_zip=False)

File F:\anaconda3\envs\torch\lib\site-packages\insightface\model_zoo\model_zoo.py:91, in get_model(name, **kwargs)
     89 if not osp.exists(model_file) and allow_download:
     90     model_file = download_onnx('models', model_file, root=root, download_zip=download_zip)
---> 91 assert osp.exists(model_file), 'model_file %s should exist'%model_file
     92 assert osp.isfile(model_file), 'model_file %s should be a file'%model_file
     93 router = ModelRouter(model_file)

AssertionError: model_file inswapper_128.onnx should exist

i am using jupyter notebook,after i download and put it into C:\Users\usrname\.insightface\models\inswapper_128,it still can't work,can anybody helps!plz!

**Comment by VikingPathak** [comment]
:heavy_check_mark: **Here are the steps to use the `inswapper_128.onnx` file:**
  * Download the model file from [this Reddit link](https://github.com/facefusion/facefusion-assets/releases/download/models/inswapper_128.onnx)
  * Check the path of the `model_zoo.py` file in the error. Navigate to the path and open the file. Just before the error line, try to print model path that the program is referring to using `print(model_file)`.
  * Place the downloaded `inswapper.onnx` file in the repective directory. It was showing just `inswapper.onnx` for me which implies root directory.
